### PR TITLE
ref: extract mark previous chapters logic into MarkPreviousChaptersUseCase

### PR DESCRIPTION
**Pull request by nonproto**
- Extracted the `markPreviousChapters` logic from `MangaViewModel` into a dedicated Use Case.
- Created `MarkPreviousChaptersUseCase.kt` which returns a Pair of `chaptersToMark` and the appropriate `ChapterMarkAction`.
- Kept the UI/View state (`chapterList`) outside of the Use Case context by passing it as a parameter from `MangaViewModel`.
- Cleaned up the DI mappings (`ChapterUseCases.kt`) to ensure `MarkPreviousChaptersUseCase` is exposed.
- Added comprehensive unit tests in `MarkPreviousChaptersUseCaseTest.kt` covering boundary conditions.

---
*PR created automatically by Jules for task [17223889218712979153](https://jules.google.com/task/17223889218712979153) started by @nonproto*

### CODE REVIEW

File: app/src/main/java/org/nekomanga/usecases/chapters/MarkPreviousChaptersUseCase.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+        val chaptersToMark = chapterList.subList(0, chapterIndex)
+        val altChapters = chapterList.subList(chapterIndex + 1, chapterList.size)
+
+        val action =
+            if (read) ChapterMarkActions.PreviousRead(true, altChapters)
+            else ChapterMarkActions.PreviousUnread(true, altChapters)
+
+        return Pair(chaptersToMark, action)
+    }
```
[the remark this answers is not in the document]
Comment (nonproto) [reply app/src/main/java/org/nekomanga/usecases/chapters/MarkPreviousChaptersUseCase.kt:26]: > medium
> 
> Using a `Pair` as a return type for a domain use case is less descriptive and can make the code harder to maintain as the meaning of `first` and `second` is not immediately clear. It is better to define a specific data class for the result to clearly communicate the intent of each property. This also allows for cleaner destructuring at the call site.
> 
> ```kotlin
>     data class Result(
>         val chaptersToMark: List<ChapterItem>,
>         val action: ChapterMarkActions,
>     )
> 
>     operator fun invoke(
>         chapterItem: ChapterItem,
>         chapterList: List<ChapterItem>,
>         read: Boolean,
>     ): Result? {
>         val chapterIndex = chapterList.indexOf(chapterItem)
>         if (chapterIndex == -1) return null
> 
>         val chaptersToMark = chapterList.subList(0, chapterIndex)
>         val altChapters = chapterList.subList(chapterIndex + 1, chapterList.size)
> 
>         val action =
>             if (read) ChapterMarkActions.PreviousRead(true, altChapters)
>             else ChapterMarkActions.PreviousUnread(true, altChapters)
> 
>         return Result(chaptersToMark, action)
>     }
> ```

@jules

File: app/src/test/java/org/nekomanga/usecases/chapters/MarkPreviousChaptersUseCaseTest.kt [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+                eu.kanade.tachiyomi.data.database.models.Chapter.create()
+                    .apply {
+                        this.id = id
+                        this.manga_id = 1L
+                        this.name = name
+                        this.url = ""
+                    }
+                    .toSimpleChapter()!!
+        )
+    }
```
[the remark this answers is not in the document]
Comment (nonproto) [reply app/src/test/java/org/nekomanga/usecases/chapters/MarkPreviousChaptersUseCaseTest.kt:20]: > medium
> 
> This helper method manually instantiates the database model `Chapter` and converts it to a domain model. It is recommended to use `mockk<ChapterItem>()` directly to simplify the test setup and avoid the overhead of implementation classes when testing domain layer extractions.
> 
> References

@jules
